scripts/validate_oas_examples.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python3
+
+"""
+The OAS file will contain JSON examples for
+- Requests
+    Endpoints that have a request body with content application/fhir+json
+- Responses
+    Endpoints that return content application/fhir+json
+
+This script looks to validate these examples.
+
+At the moment it is not validating example responses for failure scenarios.
+
+Usage:
+Running 'make publish' will execute this script
+"""
+
+import os.path
+from json import load
+from openapi_core import create_spec
+from openapi_core.validation.request.validators import RequestValidator
+from openapi_core.validation.request.datatypes import OpenAPIRequest
+from openapi_core.validation.response.validators import ResponseValidator
+from openapi_core.validation.response.datatypes import OpenAPIResponse
+from yaml import safe_load
+
+SCRIPT_LOCATION = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+REPO_ROOT = os.path.abspath(os.path.join(SCRIPT_LOCATION, ".."))
+spec_path_json = os.path.join(REPO_ROOT, "build/e-referrals-service-api.json")
+spec_path_yaml = os.path.join(REPO_ROOT, "specification/e-referrals-service-api.yaml")
+
+# List containing http methods in OAS file for sending data
+# in request body
+http_methods_data_body = ["post", "put", "patch"]
```

Comment:
This only seems to be used on the line below. Is that correct?

Reply:
Yes. I went in a different direction initially, I will merge these